ompi/include/mpi.h.in

```diff
@@ -296,13 +296,26 @@
 #            define OMPI_OMIT_MPI1_COMPAT_DECLS 0
 #            define OMPI_REMOVED_USE_STATIC_ASSERT 0
 #            define __mpi_interface_removed__(func, newfunc) __mpi_interface_deprecated__(#func " was removed in MPI-3.0.  Use " #newfunc " instead.  continuing...")
-#        elif (__STDC_VERSION__ >= 201112L)
+#        elif defined(__STDC_VERSION__) && (__STDC_VERSION__ >= 201112L)
+             // This is the C11 (or later) case, which uses
```

Comment:
The mix of C and C++ style comments hurts my eyes :).

Reply:
I was just going to reply with a 😆 and be done, but that's actually a fair point.

I think we originally didn't have any `//`-style comments in `mpi.h.in` for maximum compatibility (i.e., on the off-chance that the application's C compiler didn't support `//`-style commenting).

I see that I copied the `//`-style commenting from just a few lines down, so it looks like we introduced `//`-style commenting when these static asserts were introduced.

I think I'm going to return this to the original philosophy: no `//`-style commenting in this user-facing header file.  I'll add another commit here / re-jigger them so that the net is that everything is `/* */`-style comments.